[WIP] Add documentation for rule CA2248 (#5247)
MicrosoftDocs/visualstudio-docs: docs/code-quality/ca2248.md

---

> title: 'CA2248: Provide correct 'enum' argument to 'Enum.HasFlag''

Reviewer: ```suggestion
title: "CA2248: Provide correct 'enum' argument to 'Enum.HasFlag''"
```

Reviewer: Change to fix a rendering issue

Reviewer: We want to avoid inline styles in metadata and headings. For inline code style, we are using the backtick `

ms.date: 05/06/2020
ms.topic: reference
f1_keywords:
- CA2248
helpviewer_keywords:
- CA2248

> author: evangelink

Reviewer: ```suggestion
author: mavasani
```

> ms.author: evangelink

Reviewer: ```suggestion
ms.author: mavasani
manager: jillfra
```

> manager: 

Author: Not sure how you fill these fields.

Reviewer: We need an internal MS name for these. I'll add the suggested edits.

---

> # CA2248: Provide correct 'enum' argument to 'Enum.HasFlag'

Reviewer: ```suggestion
# CA2248: Provide correct enum argument to Enum.HasFlag
```

|||
|-|-|
|CheckId|CA2248|
|Category|Microsoft.Usage|
|Breaking change|Non-breaking|

## Cause

> The enum type passed as argument to the `HasFlag` method call is different from the calling enum type.

Reviewer: ```suggestion
The enum type passed as an argument to the `HasFlag` method call is different from the calling enum type.
```

## Rule description

> 'Enum.HasFlag' method expects the 'enum' argument to be of the same 'enum' type as the instance on which the method is invoked. If these are different 'enum' types, an unhandled exception will be thrown at runtime.

Reviewer: ```suggestion
The `Enum.HasFlag` method expects the `enum` argument to be of the same `enum` type as the instance on which the method is invoked. If these are different `enum` types, an unhandled exception will be thrown at runtime.
```

---
title: "CA2248: Provide correct enum argument to Enum.HasFlag"
ms.date: 05/06/2020
ms.topic: reference
f1_keywords:
- CA2248
helpviewer_keywords:
- CA2248

> author: mavasani

Reviewer: @mavasani Manish, we need an internal name for author. Is it best to use yours? If it's important to know whether you were the original contributor or not, we could use my name instead.

Reviewer 2: Anything works for me. However, note that this PR should be considered (and probably tagged in the title too) as a Work in progress [WIP]. @Evangelink went an extra mile to follow our new documentation guidelines for new CA rules and created a doc PR for this new rule, while the analyzer PR for this is in review: https://github.com/dotnet/roslyn-analyzers/pull/3557#issuecomment-624914045

Lets wait for that PR to be merged before reviewing/merging this.

Reviewer: Yes! Thanks, @Evangelink
